--- Python-2.7.3/Lib/ctypes/util.py.2012-08-31-144149	2012-08-31 14:38:48.000000000 -0700
+++ Python-2.7.3/Lib/ctypes/util.py	2012-08-31 14:42:46.000000000 -0700
@@ -140,7 +140,9 @@
             dump = f.read()
             rv = f.close()
             if rv == 10:
-                raise OSError, 'objdump command not found'
+                return os.path.basename(f) #  This is good for GLibc, I think,
+                                           # and a dep on binutils is big (for
+                                           # live CDs).
             f = os.popen(cmd)
             try:
                 data = f.read()
